src/clients/python/samples/walkthrough/main.py

```diff
@@ -0,0 +1,554 @@
+        id=104,
+        debits_pending=0,
+        debits_posted=0,
+        credits_pending=0,
+        credits_posted=0,
+        user_data_128=0,
+        user_data_64=0,
+        user_data_32=0,
+        ledger=1,
+        code=1,
+        timestamp=0,
+        flags=0,
+    )
+
+    account_errors = client.create_accounts([account0, account1, account2])
+    for error in account_errors:
+        if error.result == CreateAccountResult.EXISTS:
+            print(f"Batch account at {error.index} already exists.")
+        else:
+            print(f"Batch account at ${error.index} failed to create: {error.result}.")
+    # endsection:create-accounts-errors
+except:
+    raise
+
+try:
+    # section:lookup-accounts
+    accounts = client.lookup_accounts([100, 101])
+    # endsection:lookup-accounts
+except:
+    raise
+
+try:
+    # section:create-transfers
+    transfers = [Transfer(
+        id=tbid(), # TigerBeetle time-based ID.
+        debit_account_id=102,
+        credit_account_id=103,
+        amount=10,
+        pending_id=0,
+        user_data_128=0,
+        user_data_64=0,
+        user_data_32=0,
+        timeout=0,
+        ledger=1,
+        code=720,
+        flags=0,
+        timestamp=0,
+    )]
+
+    transfer_errors = client.create_transfers(transfers)
+    # Error handling omitted.
+    # endsection:create-transfers
+except:
+    raise
+
+try:
+    # section:create-transfers-errors
+    batch = [Transfer(
+        id=1,
+        debit_account_id=102,
+        credit_account_id=103,
+        amount=10,
+        pending_id=0,
+        user_data_128=0,
+        user_data_64=0,
+        user_data_32=0,
+        timeout=0,
+        ledger=1,
+        code=720,
+        flags=0,
+        timestamp=0,
+    ),
+    Transfer(
+        id=2,
+        debit_account_id=102,
+        credit_account_id=103,
+        amount=10,
+        pending_id=0,
+        user_data_128=0,
+        user_data_64=0,
+        user_data_32=0,
+        timeout=0,
+        ledger=1,
+        code=720,
+        flags=0,
+        timestamp=0,
+    ),
+    Transfer(
+        id=3,
+        debit_account_id=102,
+        credit_account_id=103,
+        amount=10,
+        pending_id=0,
+        user_data_128=0,
+        user_data_64=0,
+        user_data_32=0,
+        timeout=0,
+        ledger=1,
+        code=720,
+        flags=0,
+        timestamp=0,
+    )]
+
+    transfer_errors = client.create_transfers(batch)
+    for error in transfer_errors:
+        if error.result == CreateTransferResult.EXISTS:
+            print(f"Batch transfer at {error.index} already exists.")
+        else:
+            print(f"Batch transfer at {error.index} failed to create: {error.result}.")
+    # endsection:create-transfers-errors
+except:
+    raise
+
+try:
+    # section:no-batch
+    batch = [] # Array of transfer to create.
+    for transfer in batch:
+        transfer_errors = client.create_transfers(transfer)
+        # Error handling omitted.
```

Comment:
list comprehension would be more idiomatic here

Reply:
It would be normally, but I think the straight for loop is better for showing the "don't do this" nature of the example?

Reply:
(although, that code won't actually work if batches is non-empty, because create_transfers expects a list!)